source/openapi-admin-v3.yaml

```diff
@@ -3879,7 +3877,122 @@ components:
+    Role:
+      type: object
+      description: A data access role that assigns read and write permissions to a user for each document
+      properties:
+        _id:
+          type: string
+          description: The role's unique ObjectId identifier.
+        name:
+          type: string
+          description: The role's name.
+        apply_when:
+          description: A rule expression that evaluates to `true` when this role applies to a user for a specific document.
+          $ref: "#/components/schemas/Expression"
+        write:
+          description: A rule expression that evaluates to `true` if the role has permission to add, modify, or remove all fields in the document.
+          $ref: "#/components/schemas/Expression"
+        insert:
+          description: A rule expression that evaluates to `true` if the role has permission to insert a new document into the collection.
+          $ref: "#/components/schemas/Expression"
+        delete:
+          description: A rule expression that evaluates to `true` if the role has permission to delete a document from the collection.
+          $ref: "#/components/schemas/Expression"
+        search:
+          description: A rule expression that evaluates to `true` if the role has permission to search the collection using [Atlas Search](https://www.mongodb.com/docs/atlas/atlas-search/).
+          $ref: "#/components/schemas/Expression"
+        fields:
+          type: object
+          description: Field-level rules for any fields that are not explicitly configured in the role's `properties`.
+          $ref: "#/components/schemas/NestableFieldLevelPermissions"
+        additional_fields:
+          type: object
+          description: Field-level rules for any fields that are not explicitly configured in the role's `properties`.
+          $ref: "#/components/schemas/FieldLevelPermissions"
+    NestableFieldLevelPermissions:
+      description: An object that maps document field names to read and write permissions for the field and its children.
+      additionalProperties:
+        allOf:
+          - $ref: "#/components/schemas/FieldLevelPermissions"
+          - properties:
+              fields:
+                $ref: "#/components/schemas/NestableFieldLevelPermissions"
+    FieldLevelPermissions:
+      type: object
+      description: Read and write permissions for a field.
+      required:
+        - read
+        - write
+      properties:
+        read:
+          description: A rule expression that evaluates to `true` if the role can read the field's value.
+          $ref: "#/components/schemas/Expression"
+        write:
+          description: A rule expression that evaluates to `true` if the role can write the field's value.
+          $ref: "#/components/schemas/Expression"
+    Filter:
+      type: object
+      description: A filter that conditionally adds conditions to and projects incoming operations before they run.
```

Comment:
Nit: "conditionally adds conditions to" was tough for me to parse. What about something like:

> When matched, a filter adds conditions to and projects...